platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/internal/extensibility/DeprecatedBooleanPropertyIntegrationTest.groovy

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.extensibility
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class DeprecatedBooleanPropertyIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
I think it would be interesting to see similar tests verifying groovy 4's behavior and being able to compare our deprecation warnings vs whether groovy 4 recognizes it as a property

Reply:
I'm not really sure what the value of this test would be. Are you saying I should make a variant of each deprecating-emitting test that uses Groovy 4 and expects to fail, and each one that doesn't emit a deprecation should use Groovy 4 and work?

Reply:
I guess I want to be convinced that our deprecations align with the actual behavior of groovy 4

Reply:
There are 4 getter types

Boolean isProperty
Boolean getProperty
boolean isProperty
boolean getProperty

For each, I would expect that either:
Groovy 4 recognizes it as a property
Or
We emit a deprecation

We don't necessarily need to codify this as a test if we're confident we've covered all the bases (esp if it won't be quick to implement)

Just thought it might be a robust way of validating that we are deprecating the right thing